### PR TITLE
Update the DJ app in support of App Mesh Controller for k8s GA release.

### PR DESCRIPTION
**Pull request by jlbutler**
*Issue #, if available:* addresses #304 

*Description of changes:*

This changeset updates the DJ App in support of the AWS App Mesh Controller for K8s GA. It's been updated to use v1beta2 version of the API.

It has been refactored to go along with updates on the eksworkshop.com 'Service Mesh with AWS App Mesh' module.  (https://github.com/aws-samples/eks-workshop/pull/818).

The previous version of the app has been stashed in a legacy directory for now, for those that wish to use it. It can be deprecated at some point in the future.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/apps/djapp/legacy/4_create_initial_mesh_components/nodes_representing_virtual_services.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ spec:
     dns:
       hostName: metal.prod.svc.cluster.local
```
Comment (fawadkhaliq) [inline examples/apps/djapp/legacy/4_create_initial_mesh_components/nodes_representing_virtual_services.yaml:14]: `hostName` -> `hostname`  

Needs to be changed in all the places

File: examples/apps/djapp/legacy/4_create_initial_mesh_components/virtual-services.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-apiVersion: appmesh.k8s.aws/v1beta1
+apiVersion: appmesh.k8s.aws/v1beta2
 kind: VirtualService
 metadata:
   name: jazz.prod.svc.cluster.local
```
Comment (fawadkhaliq) [inline examples/apps/djapp/legacy/4_create_initial_mesh_components/virtual-services.yaml:4]: `VirtualRouter` is no longer part of `VirtualService`

File: examples/apps/djapp/legacy/4_create_initial_mesh_components/nodes_representing_virtual_services.yaml (hunk 1 above)
Comment (jlbutler) [reply examples/apps/djapp/legacy/4_create_initial_mesh_components/nodes_representing_virtual_services.yaml]: This was cruft leftover from notes I made early on. Fixed. Everything in legacy is the old version of the app.

File: examples/apps/djapp/legacy/4_create_initial_mesh_components/virtual-services.yaml (hunk 2 above)
Comment (jlbutler) [reply examples/apps/djapp/legacy/4_create_initial_mesh_components/virtual-services.yaml]: Same issue, fixed.

File: examples/apps/djapp/legacy/4_create_initial_mesh_components/mesh.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+---
```
Comment (fawadkhaliq) [inline examples/apps/djapp/legacy/4_create_initial_mesh_components/mesh.yaml:1]: Does this work with legacy crds?

Comment (jlbutler) [reply examples/apps/djapp/legacy/4_create_initial_mesh_components/mesh.yaml]: more cruft, good catch! sorted now, i checked master out and diff'd against all originals. these were just placeholders for myself to fix - before i just wrote all new yaml for v1beta2. sorted now, legacy is back to square, with the exception of adding 'namespace' in some. thanks!

**Pull request merged**